lib/ui/src/components/preview/tools/menu.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { IconButton, Icons, Separator } from '@storybook/components';
+import { Consumer, Combo } from '@storybook/api';
+import { Addon } from '@storybook/addons';
+
+const menuMapper = ({ api, state }: Combo) => ({
+  isVisible: state.layout.showNav,
+  toggle: api.toggleNav,
+});
+
+export const menuTool: Addon = {
+  title: 'menu',
+  id: 'menu',
+  match: ({ viewMode }) => viewMode === 'story',
+  render: () => (
+    <>
+      <Consumer filter={menuMapper}>
+        {({ isVisible, toggle }) =>
+          isVisible && (
+            <>
+              <IconButton
+                aria-label="Show sidebar"
+                key="menu"
+                onClick={() => toggle()}
```

Comment:
```suggestion
                onClick={toggle}
```

Reply:
This didn't seem to work well on my editor.

Reply:
Is it a linting error? What are you seeing?
This works for me locally. [Click here to read more on passing arrow functions](https://reactjs.org/docs/faq-functions.html#arrow-function-in-render).
![image](https://user-images.githubusercontent.com/17681528/123862175-a3affa80-d8ed-11eb-86c8-ac555b748688.png)

Reply:
It is a ts(2322) error.
![screenshot](https://user-images.githubusercontent.com/56366807/123863972-e21bd880-d932-11eb-94bc-1eaa391a0dc9.png)

Reply:
try `toggle as any` and see if that cures the issue 😈 if it doesn't, don't worry about it. this is a negligible request 👍

Reply:
Yes, that was it! Thank you!